### PR TITLE
Add option --from-blockheight

### PR DESCRIPTION
**Pull request by lvaccaro**
Add option `--from-blockheight` for lightningd to define the first block to add at startup time.
If not specify, or set to default value -1, lightningd uses the current strategy.
In according to `setup_topology()`, block download starts one before the block selected in `--from-blockheight`.

Help:
```bash
$ lightingd --help
...
--max-locktime-blocks <arg>    Maximum blocks a peer can lock up our funds (default: 432)
--from-blockheight <arg>       Add blocks from blockchain height (default: -1)
--anchor-onchain <arg>         Blocks before we give up on pending anchor transaction (default: 100)
...
```
Example:
```bash
$ lightingd --from-blockheight 500000
...
2018-02-13T16:02:05.745Z lightningd(28362): Adding block 499999: 0000000000000000007962066dcd6675830883516bcf40047d42740a85eb2919
...
2018-02-13T16:02:05.944Z lightningd(28362): Adding block 500000: 00000000000000000024fb37364cbf81fd49cc2d51c09c75c35433c3a1945d04
2018-02-13T16:02:40.595Z lightningd(28442): Adding block 500001: 0000000000000000005c9959b3216f8640f94ec96edea69fe12ad7dee8b74e92
```

Resolve the following issue:
1. send satoshi in a lightning wallet created by `lightning-cli newaddr` (for example at height 500000)
2. check incoming transaction: `lightning-cli listfunds` show outputs available to spent
3. wait 102 blocks
4. restart lightningd (in log: first block added is 100-1 blocks before the current blockchain height = 500001)
5. `lightning-cli listfunds` shows no outputs

Start lightningd with the option `--from-blockheight 500000` resolves this issue and `listfunds` shows outputs available to spent.

### CODE REVIEW

**Comment by rustyrussell** [comment]
Hi!  Thanks for this, could you also look at #940 which has feedback from @cdecker and me?

I think the right fix for this is to adjust the low watermark based on incoming transactions; @cdecker was looking at saving some of this information to the db, but perhaps we should just put a wallet low-watermark in there for now?

File: lightningd/lightningd.h [hunk 1]
```diff
@@ -26,6 +26,9 @@ struct config {
 	/* How long do we let them lock up our funds? (blocks) */
 	u32 locktime_max;
 
+	/* Add block from blockheight (-1 to use default chaintopology strategy). */
+	s32 from_blockheight;
```
Comment (cdecker) [inline lightningd/lightningd.h]: This is probably a year-43k bug in the making :wink:

File: lightningd/lightningd.c [hunk 2]
```diff
@@ -325,6 +325,11 @@ int main(int argc, char *argv[])
 
 	peer_first_blocknum = wallet_channels_first_blocknum(ld->wallet);
 
+	if (ld->config.from_blockheight > -1){
+		// Get_init_blockhash() go back of 100 blocks
+		peer_first_blocknum = ld->config.from_blockheight + 100;
```
[the remark this answers is not in the document]
Comment (lvaccaro) [reply lightningd/lightningd.c]: [get_init_blockhash](https://github.com/ElementsProject/lightning/blob/8f7a19d1a3c5967663fe8f850b24e75a2a259339/lightningd/chaintopology.c#L475) starts 100 blocks behind the selected block. If it is more safe to always consider 100 blocks before the selected input block ( so the first "Adding block", in the log, starts 100 blocks before the specified blockheight in the option), I will remove this sum.

File: lightningd/lightningd.h (hunk 1 above)
Comment (cdecker) [reply lightningd/lightningd.h]: Just kidding :-)

File: lightningd/lightningd.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (cdecker) [reply lightningd/lightningd.c]: The `+` suggestes to me that if I specify height 500 this would start at 600 which is rather uexpected to me.

**Comment by rustyrussell** [comment]
Um, we're getting distracted :)  The real fix is to put wallet utxss into the db; we can do the minimal thing if we want (just keep the low watermark in db vars) but I wanted to know if @cdecker had other plans to put wallet utxos into the db?

**Comment by cdecker** [comment]
Well we have our own UTXOs (except the `onchaind` tracked ones) in the `outputs` table. We can of course hook into the `onchaind` watches to add those as well. Any low water mark would need to be below the `onchaind` tracking anyway so I think I'll just do that. Alternatively we can track all P2WSH outputs as well, so we can feed txout lookups from a table instead of calling out to `bitcoind`, but I'm not sure yet whether that overhead is worth it.

**Pull request closed without merging**